aiidalab_widgets_base/process.py

```diff
@@ -701,29 +708,31 @@ def __init__(self, callbacks=None, on_sealed=None, timeout=None, **kwargs):
 
         super().__init__(**kwargs)
 
-    @traitlets.observe("process")
+    @traitlets.observe("process_uuid")
     def _observe_process(self, change):
-        process = change["new"]
-        if process is None or process.id != getattr(change["old"], "id", None):
-            with self.hold_trait_notifications():
-                with self._monitor_thread_lock:
-                    # stop thread (if running)
-                    if self._monitor_thread is not None:
-                        self._monitor_thread_stop.set()
-                        self._monitor_thread.join()
-
-                    # start monitor thread
-                    if process is not None:
-                        self._monitor_thread_stop.clear()
-                        process_id = getattr(process, "id", None)
-                        self._monitor_thread = Thread(
-                            target=self._monitor_process, args=(process_id,)
-                        )
-                        self._monitor_thread.start()
-
-    def _monitor_process(self, process_id):
-        assert process_id is not None
-        process = load_node(process_id)
+        """When the process_uuid is changed, stop the previous
+        monitor if exist. Start a new one in thread."""
+        process_uuid = change["new"]
+
+        if process_uuid is None:
+            return
+
+        # stop thread (if running)
```

Comment:
@unkcpz I think the thread needs to be stopped when the process_uuid is set to None